processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -310,7 +331,10 @@ public void advance()
                       if (Thread.interrupted()) {
                         throw new QueryInterruptedException();
                       }
-                      cursorOffset.increment();
+
+                      do {
+                        cursorOffset.increment();
+                      } while (!isDone() && !filterMatcher.matches());
```

Comment:
i.e. how much slower is it when there is no filter, or a an always true filter compared to the previous implementation.